### PR TITLE
Option to build tests

### PR DESCRIPTION
**Pull request by Alan-Jowett**
This pull request introduces a new option to conditionally build and run tests in the CI workflow and modifies the CMake configuration to support this option. The changes are primarily focused on enhancing the build process and improving the flexibility of the CI pipeline.

### CI Workflow Enhancements:
* [`.github/workflows/build.yml`](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R21): Added a `build_tests` matrix to conditionally build and run tests. Updated the build and test steps to respect this new matrix parameter. [[1]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R21) [[2]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L40-R49) [[3]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R58) [[4]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L69-R77)

### CMake Configuration Updates:
* [`CMakeLists.txt`](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR8-R9): Introduced a new option `VERIFIER_ENABLE_TESTS` to control the inclusion of test-related targets and dependencies. Wrapped test-related configuration and target definitions in conditional statements based on this option. [[1]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR8-R9) [[2]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR18-R25) [[3]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR98) [[4]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR107) [[5]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR124-R133) [[6]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR151-R161) [[7]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR179) [[8]](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR200)

## Summary by CodeRabbit

- **New Features**
	- Introduced a flexible build configuration that allows conditional execution of tests during the CI process for both Ubuntu and Windows builds.
	- Added a new option to enable or disable tests in the build process.

- **Bug Fixes**
	- Enhanced the build process by ensuring that test executables are only included when the appropriate option is enabled.

- **Documentation**
	- Updated documentation to reflect the new testing configuration options available in the build process.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
@coderabbitai full review

**Comment by elazarg** [comment]
@Alan-Jowett feel free to dismiss coderabbit comments you deem irrelevant or wrong

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -66,8 +70,9 @@ jobs:
       - name: Build
         run: |
           mkdir build
-          cmake -B build
+          cmake -B build -DVERIFIER_ENABLE_TESTS=${{matrix.build_tests}}
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply .github/workflows/build.yml:73]: Other locations have if: clauses with the operand not wrapped in ${{}} so this is consistent with the general style.

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,12 +15,14 @@ FetchContent_Declare(GSL
...
 get_target_property(GSL_INCLUDE_DIRS Microsoft.GSL::GSL INTERFACE_INCLUDE_DIRECTORIES)
 
+if (VERIFIER_ENABLE_TESTS)
 FetchContent_Declare(Catch2
         GIT_REPOSITORY "https://github.com/catchorg/Catch2.git"
         GIT_TAG "v3.7.1"
         GIT_SHALLOW ON
 )
 FetchContent_MakeAvailable(Catch2)
+endif()
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply CMakeLists.txt:25]: Existing code. Not modified in this PR.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -37,13 +38,15 @@ jobs:
       - name: Build
         run: |
           mkdir build
-          cmake -B build -DCMAKE_BUILD_TYPE=${{env.BUILD_CONFIGURATION}}
+          cmake -B build -DCMAKE_BUILD_TYPE=${{env.BUILD_CONFIGURATION}} -DVERIFIER_ENABLE_TESTS=${{matrix.build_tests}}
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply .github/workflows/build.yml:41]: Single line is consistent with the style of this YML document.

**Comment by elazarg** [comment]
Two comments:
1. Is it possible/sensible to give names to the tests option instead of true/false?
2. If you do any other change, please add new line at the end of CMakeLists.txt

**Review by elazarg — APPROVED** [review]

**Pull request merged**